Library/Homebrew/download_strategy.rb

```diff
@@ -449,7 +449,7 @@ def resolve_url_basename_time_file_size(url, timeout: nil)
     return @resolved_info_cache[url] if @resolved_info_cache.include?(url)
 
     if (domain = Homebrew::EnvConfig.artifact_domain)
-      url = url.sub(%r{^((ht|f)tps?://)?}, "#{domain.chomp("/")}/")
+      url = url.sub(%r{^((ht|f)tps?://ghcr.io/)?}, "#{domain.chomp("/")}/")
```

Comment:
```suggestion
      url = url.sub(%r{^(https?://#{GitHubPackages::URL_DOMAIN}/)?}, "#{domain.chomp("/")}/")
```

Reply:
Agree, with 2 additional changes:
1. Added the `o` flag to handle RuboCop error: https://msp-greg.github.io/rubocop-performance/RuboCop/Cop/Performance/ConstantRegexp.html
2. URL_DOMAIN constant became public